expr/func_substr.go

```diff
@@ -0,0 +1,86 @@
+package expr
+
+import (
+	"strings"
+
+	"github.com/grafana/metrictank/api/models"
+)
+
+type FuncSubstr struct {
+	in    GraphiteFunc
+	start int64
+	stop  int64
+}
+
+func NewSubstr() GraphiteFunc {
+	return &FuncSubstr{}
+}
+
+func NewAliasByMetric() GraphiteFunc {
+	return &FuncSubstr{start: -1, stop: 0}
+}
+
+func (s *FuncSubstr) Signature() ([]Arg, []Arg) {
+	if s.start != 0 || s.stop != 0 {
+		return []Arg{
+			ArgSeriesList{val: &s.in},
+		}, []Arg{ArgSeriesList{}}
+	}
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+		ArgInt{val: &s.start, opt: true, key: "start"},
+		ArgInt{val: &s.stop, opt: true, key: "stop"},
+	}, []Arg{ArgSeriesList{}}
+}
+
+func (s *FuncSubstr) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncSubstr) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
+	series, err := s.in.Exec(cache)
+	if err != nil {
+		return nil, err
+	}
+
+	out := make([]models.Series, len(series))
+	for i, serie := range series {
+		left := strings.LastIndex(serie.Target, "(") + 1
+		right := strings.Index(serie.Target, ")")
+		if right < 0 {
+			right = len(serie.Target)
+		}
+		cleanName := serie.Target[left:right]
+		cleanName = strings.SplitN(cleanName, ",", 2)[0]
```

Comment:
This is another one that likely should be updated in graphite-web to use `_getFirstPathExpression` for consistency with the way we get nodes in `aliasByNode` etc.  This implementation is consistent with what's in graphite-web today though.